### PR TITLE
chore(query): make native format respect data cache and merge io results

### PR DESCRIPTION
**Pull request by sundy-li**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Summary about this PR

- Closes #issue

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/12835)

### CODE REVIEW

**Comment by sundy-li** [comment]
askbend:summary

File: src/query/storages/common/cache/src/providers/table_data_cache.rs [hunk 1]
```diff
@@ -103,7 +104,7 @@ impl CacheAccessor<String, Vec<u8>, DefaultHashBuilder, Count> for TableDataCach
         }
     }
 
-    fn put(&self, k: String, v: Arc<Vec<u8>>) {
+    fn put(&self, k: String, v: Arc<Bytes>) {
```
Comment (Xuanwo) [inline src/query/storages/common/cache/src/providers/table_data_cache.rs:107]: Maybe we can use `Bytes` directly instead of `Arc<Bytes>`?

**Review by drmingdrmer — COMMENTED** [review]
> Reviewed 11 of 22 files at r1, 10 of 13 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/datafuselabs/databend/12835)* status: 21 of 27 files reviewed, 1 unresolved discussion (waiting on @sundy-li and @Xuanwo)

**Review by Xuanwo — APPROVED** [review]

**Comment by BohuTANG** [comment]
Wait the benchmark work and do a check.

**Pull request merged**